### PR TITLE
Use type-system-web package for types

### PR DESCRIPTION
**Pull request by thehabbos007**
## 🌟 What is the purpose of this PR?

This PR makes use of the new Block Protocol `@blockprotocol/type-system-web` package containing new type definitions for the Graph API

## 🔗 Related links

- Related to #977 

## 🔍 What does this change?

- Replaces `@hashintel/hash-graph-client` usages with `@blockprotocol/type-system-web` in places where it makes sense
  - GraphQL types now use these new types as scalars instead of the ones generated from the Graph client
-  The existing example usage of the BP functions have a small example use of the new type system package

## ⚠️ Known issues

- We are currently using the `*-web` variant of the package for both front and backend, which is not ideal, and will be unified soon.
- The packages are being used through https://gitpkg.vercel.app/, which may or may not be a solid dependency for our packages.

## 🐾 Next steps

-  Unifying the package.
-  Making use of the new types for creating UI elements.
-  Using the new package as a yarnpkg/npm dependency instead of gitpkg.

## 🛡 What tests cover this?

- None

## ❓ How to test this?

The new package should be available in the frontend, but no observed changes otherwise

### CODE REVIEW

File: packages/hash/api/src/model/ontology/property-type.model.ts [hunk 1]
```diff
@@ -62,7 +63,11 @@ export default class {
     return persistedPropertyTypes.map(
       (persistedPropertyType) =>
         new PropertyTypeModel({
-          schema: persistedPropertyType.inner,
+          /**
+           * @todo and a warning, these type casts are here to satisfy the type system package.
```
Comment (Alfred-Mountfield) [inline packages/hash/api/src/model/ontology/property-type.model.ts:67]: I'm not certain about this message. I think it might be giving the wrong idea. Is this really to satisfy the type system package? Or is this because our generated OpenAPI client isn't good enough?

File: packages/hash/api/src/model/util.ts [hunk 2]
```diff
@@ -148,7 +150,7 @@ export const generateWorkspacePropertyTypeSchema = (params: {
     namespaceUri: workspaceTypesNamespaceUri,
     title: params.title,
     kind: "propertyType",
-  }),
```
Comment (Alfred-Mountfield) [inline packages/hash/api/src/model/util.ts:151]: Can we actually replace this cast now? I think we should modify `generateSchemaVersionedUri` to return the correct type. I'm guessing we can't easily call `isVersionedUri` because of the node/web problem? 🙃

File: packages/hash/frontend/src/pages/type-editor/index.page.tsx [hunk 3]
```diff
@@ -45,27 +46,30 @@ const ExampleUsage = ({ accountId }: { accountId: string }) => {
...
-              kind: "propertyType",
-              $id: propertyUri,
-              title: "Name",
-              oneOf: [
-                {
-                  $ref: "https://blockprotocol.org/@blockprotocol/types/data-type/text/v/1",
-                },
-              ],
+      if (isVersionedUri(propertyUri)) {
+        // now propertyUri is typed as VersionedUri from the type system
```
Comment (Alfred-Mountfield) [inline packages/hash/frontend/src/pages/type-editor/index.page.tsx:50]: I think this comment is a little confusing if I'm honest

Comment (Alfred-Mountfield) [reply packages/hash/frontend/src/pages/type-editor/index.page.tsx:50]: If you're trying to indicate that this is using methods from the type system then maybe rephrase as something like "use the type guard from the package"

**Review by Alfred-Mountfield — COMMENTED** [review]
I appreciate the warnings being put in, I think we might want to better capture the problems we're facing, and the conditions under which they change. As in, if we unify the type-system-node/web package that would remove some of the problems. If we manage to improve the types generated by the OpenAPI client, or manage to swap them out, then that would also remove some of the problems.

File: packages/hash/api/src/graphql/resolvers/ontology/model-mapping.ts [hunk 4]
```diff
@@ -24,6 +24,7 @@ export const propertyTypeModelToGQL = (
 ): PersistedPropertyType => ({
   accountId: propertyType.accountId,
   propertyTypeVersionedUri: propertyType.schema.$id,
+  /** @todo and a warning, these type casts are here to satisfy the type system package */
```
[the remark this answers is not in the document]
Comment (thehabbos007) [reply packages/hash/api/src/graphql/resolvers/ontology/model-mapping.ts:27]: Oh, i had bubbled the type cast up, this comment was left after that. Removing now

File: packages/hash/api/src/model/util.ts (hunk 2 above)
Comment (thehabbos007) [reply packages/hash/api/src/model/util.ts]: Adding `as const` in the `generateSchemaVersionedUri` made the cast redundant :+1: 

c5fdb196fa10666131dd10aff5d7973af9cce7ba

File: packages/hash/frontend/src/pages/type-editor/index.page.tsx (hunk 3 above)
Comment (thehabbos007) [reply packages/hash/frontend/src/pages/type-editor/index.page.tsx]: I think it's an unnecessary comment anyway, I'll remove it 250b06ad829bdf3d603d00be256a59d31b02c396

File: packages/hash/api/src/model/ontology/property-type.model.ts (hunk 1 above)
Comment (thehabbos007) [reply packages/hash/api/src/model/ontology/property-type.model.ts]: I know it's quite verbose/long but changed it in f5fb24ad45ccbe41425b7b56c9271eb7a7ce25df. I want to have these warnings as a blaring piece until we sort it out.

I think the shape is actually matching, but we're not asserting stuff like the `$id`s and `$refs` matching the template literal for instance.

Comment (Alfred-Mountfield) [reply packages/hash/api/src/model/ontology/property-type.model.ts]: Much better, thank you

**Review by Alfred-Mountfield — APPROVED** [review]
Linting is failing, happy to re-approve when ready :)

File: packages/hash/api/package.json [hunk 5]
```diff
@@ -23,6 +23,7 @@
     "@aws-sdk/s3-presigned-post": "3.38.0",
     "@aws-sdk/s3-request-presigner": "3.38.0",
     "@blockprotocol/core": "0.0.8",
+    "@blockprotocol/type-system-web": "https://gitpkg.now.sh/blockprotocol/blockprotocol/packages/%40blockprotocol/type-system-web?c0a430d",
```
Comment (Alfred-Mountfield) [inline packages/hash/api/package.json:26]: I think we should temporarily just whitelist this domain for dependencies, with a TODO. We're in a dev branch 🤷
